src/layout/GeometryMap/GeometryMapComponent.tsx

```diff
@@ -0,0 +1,193 @@
+      attribution: 'Data © <a href="https://www.kartverket.no/">Kartverket</a>',
+    },
+  ];
+
+  const coordinates: any = [];
+  const labels: any = [];
+
+  const formData = FD.useFreshBindings(dataModelBindings, 'raw');
+  if (Array.isArray(formData['array'])) {
+    for (const obj of formData['array']) {
+      const arrayBinding = dataModelBindings['array'];
+      const textBinding = dataModelBindings.label.replace(`${arrayBinding}.`, '');
+      const wktStringBinding = dataModelBindings.wkt.replace(`${arrayBinding}.`, '');
+      coordinates.push(dot.pick(wktStringBinding, obj));
+      labels.push(dot.pick(textBinding, obj));
+    }
+  } else {
+    coordinates.push(formData['wkt']);
+    labels.push(formData['label']);
+  }
+
+  const [inputCoords, geometryType] = findCoordinates(coordinates);
+
+  const polyCenter = findPolygonCenter(inputCoords[0][0]);
+
+  const DefaultCenterLocation: Location = {
+    latitude: 64.888996,
+    longitude: 12.8186054,
+  };
+
+  const center = polyCenter
+    ? polyCenter
+    : centerLocation
+      ? locationToTuple(centerLocation)
+      : locationToTuple(DefaultCenterLocation);
+
+  const [map, setMap] = useState<LeafletMap | null>(null);
+
+  return (
+    <div className={`geometry-map-component${isValid ? '' : ' validation-error'}`}>
+      <MapContainer
+        className={classes.map}
+        center={center}
+        ref={setMap}
+        zoom={zoom ? zoom : polyCenter ? 4 : 16}
+        dragging={!readOnly}
+        attributionControl={false}
+      >
+        {layers.map((layer, i) => (
+          <TileLayer
+            key={i}
+            url={layer.url}
+            attribution={layer.attribution}
+            subdomains={layer.subdomains ? layer.subdomains : []}
+            opacity={readOnly ? 0.5 : 1.0}
+          />
+        ))}
+        <AttributionControl prefix={false} />
+
+        {inputCoords.map(
+          (coords, i) =>
+            geometryType[i] === 'polygon' ?? (
+              <Polygon
+                key={i}
+                positions={coords}
+              >
+                <Tooltip>{labels !== undefined ? <span>{JSON.stringify(labels[i])}</span> : <div></div>}</Tooltip>
+              </Polygon>
+            ),
+        )}
+      </MapContainer>
+    </div>
+  );
+}
+
+export function parseLocation(locationString: string | undefined): Location | undefined {
+  if (!locationString) {
+    return undefined;
+  }
+  const latLonArray = locationString.split(',');
+  if (latLonArray.length != 2) {
+    window.logErrorOnce(`Invalid location string: ${locationString}`);
+    return undefined;
+  }
+  const latString = latLonArray[0];
+  const lonString = latLonArray[1];
+  const lat = parseFloat(latString);
+  const lon = parseFloat(lonString);
+  if (isNaN(lat) || isNaN(lon)) {
+    window.logErrorOnce(`Invalid location string: ${locationString}`);
+    return undefined;
+  }
+  return {
+    latitude: lat,
+    longitude: lon,
+  } as Location;
+}
```

Comment:
Er denne duplisert fra `MapComponent.tsx`? Kunne kanskje laget en `mapUtils.ts` fil under `src/utils/` for slike ting

Reply:
Den var tatt med fra MapComponent ja, innså nå at jeg aldri tok den i bruk heller da lesing endte med å gjøres på annen måte, så har fjernet det fra GeoMap nu!